src/jats/importer/jats-parser-utils.ts

```diff
@@ -56,24 +62,17 @@ const updateNodeID = (
   replacements: Map<string, string>,
   warnings: string[]
 ) => {
-  if (node.type === schema.nodes.inline_equation) {
+  if (nonJSONSchemaTypes.has(node.type.name)) {
```

Comment:
Yeah, the only reason i can think of is because we didn't want to generate ids for some of the nodes that didn't have an objectType (e.g. body, backmatter). But I also don't know why we wouldn't want to generate ids for those nodes. 
I updated the code.

Reply:
Actually the reason for this was that `generateNodeID` didn't generate ids for non json schema types; it uses `nodeTypesMap` to generate the ids.

Reply:
In that case let's update `generateNodeID` to support this instead of having the code in the util.